File: squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution(object):
    def sortedSquares(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        # First, square all numbers in nums for O(n) time.
        for i, num in enumerate(nums):
            nums[i] = num * num
            
        # Then, sort the nums array in presumably O(n logn) time.
        return sorted(nums)
        

File: squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    
    public int[] sortedSquares(int[] nums) {
        
        // First, square all numbers in nums for O(n) time.
        for (int i = 0; i < nums.length; i++) {
            nums[i] = nums[i] * nums[i];
        }
            
        // Then, sort the nums array in presumably O(n logn) time.
        Arrays.sort(nums);
        return nums;
        
    }
    
}